File: CS/09_Forms/FillXFAFields.cs
using Spire.Pdf;
using Spire.Pdf.Widget;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace FillXFAFields
{
    public partial class Form1 : Form
    { 
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Load a Pdf file
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\XFASample.pdf");

            PdfFormWidget formWidget = doc.Form as PdfFormWidget;
            List<XfaField> xfafields = formWidget.XFAForm.XfaFields;

            // Iterate through each XFA field in the form
            foreach (XfaField xfaField in xfafields)
            {
                if (xfaField is XfaTextField)
                {
                    XfaTextField textField = xfaField as XfaTextField;
                    textField.Value = "E-iceblue";
                }
                if (xfaField is XfaDateTimeField)
                {
                    XfaDateTimeField datetimeField = xfaField as XfaDateTimeField;
                    datetimeField.Value = DateTime.Now.ToString();
                }
                if (xfaField is XfaCheckButtonField)
                {
                    XfaCheckButtonField checkButtonField = xfaField as XfaCheckButtonField;
                    checkButtonField.Checked = true;
                }
                if (xfaField is XfaChoiceListField)
                {
                    XfaChoiceListField choiceListField = xfaField as XfaChoiceListField;
                    choiceListField.SelectedItem = choiceListField.Items[0];
                    choiceListField.SelectedItems.Add("NewItem");
                    
                }
                if (xfaField is XfaDoubleField)
                {
                    XfaDoubleField doubleField = xfaField as XfaDoubleField;
                    doubleField.Value = 2.14;
                }

            }

            // Save the result pdf file
            doc.SaveToFile("FillXfaField.pdf", FileFormat.PDF);

            PDFDocumentViewer("FillXfaField.pdf");
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/09_Forms/FillXFAFields.vb
Imports Spire.Pdf
Imports Spire.Pdf.Widget
Imports System.ComponentModel
Imports System.Text

Namespace FillXFAFields
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Load a Pdf file
			Dim doc As New PdfDocument()
			doc.LoadFromFile("..\..\..\..\..\..\Data\XFASample.pdf")

			Dim formWidget As PdfFormWidget = TryCast(doc.Form, PdfFormWidget)
			Dim xfafields As List(Of XfaField) = formWidget.XFAForm.XfaFields

			' Iterate through each XFA field in the form
			For Each xfaField As XfaField In xfafields
				If TypeOf xfaField Is XfaTextField Then
					Dim textField As XfaTextField = TryCast(xfaField, XfaTextField)
					textField.Value = "E-iceblue"
				End If
				If TypeOf xfaField Is XfaDateTimeField Then
					Dim datetimeField As XfaDateTimeField = TryCast(xfaField, XfaDateTimeField)
					datetimeField.Value = Date.Now.ToString()
				End If
				If TypeOf xfaField Is XfaCheckButtonField Then
					Dim checkButtonField As XfaCheckButtonField = TryCast(xfaField, XfaCheckButtonField)
					checkButtonField.Checked = True
				End If
				If TypeOf xfaField Is XfaChoiceListField Then
					Dim choiceListField As XfaChoiceListField = TryCast(xfaField, XfaChoiceListField)
					choiceListField.SelectedItem = choiceListField.Items(0)
					choiceListField.SelectedItems.Add("NewItem")
				End If
				If TypeOf xfaField Is XfaDoubleField Then
					Dim doubleField As XfaDoubleField = TryCast(xfaField, XfaDoubleField)
					doubleField.Value = 2.14
				End If

			Next xfaField

			' Save the result pdf file
			doc.SaveToFile("FillXfaField.pdf", FileFormat.PDF)

			PDFDocumentViewer("FillXfaField.pdf")
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
